### PR TITLE
Update Data Pipeline vignette

### LINKED ISSUE #1561: QC: fully document both "analytics" and "reporting" data models in a vignette
## QC Details
- Analytics data model applies data generated using to KRI/Country workflows (See existing data model refactor issue)
   - `dfInput`
   - `dfTransofrmed`
   - `dfAnalyzed`
   - `dfBounds`
   - `dfFlagged`
   - `dfSummary`
- Reporting data model applies to data used by charting and reporting functions
  - `dfSummary`
  - `dfBounds`
  - `dfSite`
  - `dfStudy`
  - `dfMetrics`

Specify all columns along with type and whether it is optional or required. 

Fine to throw this in a discussion or wiki page for now, we can convert to vignette and/or data files later.

### LINKED ISSUE #1580: QC: Update data pipeline vignette
## QC Details
Update `DataPipeline.md` vignette to reflect v2 workflow and functions

### PR DESCRIPTION
**Pull request by lauramaxwell**
## Overview
update the data pipeline vignette. includes data model wiki as an appendix.

closes #1580
closes #1561

## Test Notes/Sample Code

Notes:

### CODE REVIEW

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:102]:
```suggestion
 - `timeSeriesContinuousMetricJS`: A time series chart using JavaScript with metric on the y-axis.
 - `timeSeriesContinuousScoreJS`: A time series chart using JavaScript with score on the y-axis.
```
Previous charts were in "metric then score" order.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:123]:
```suggestion
# Make Workflow Lists
```

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:167]:
```suggestion
# Create Charts for all metrics
```

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:177]:
```suggestion
)) %>% setNames(metrics)
```
Since the parentheses are opened on a single line, it's less confusing if they're also closed on a single line.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:188]:
```suggestion
Additional examples are provided in the [Cookbook Vignette](Cookbook.html).
```
Or link in the same way as the previous one, if the magic to recode the link is intentional.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:208]:
```suggestion
  - Usage: The base data.frame for all Analytics workflows. Feeds into the `Transform_*()` functions.
```
Standardizing references to function families.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:226]:
```suggestion
  - Usage: Convert from input data format to needed format to derive KRI for an Assessment via the `Analyze_*()` functions.
```

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:245]:
```suggestion
  - Usage: Prepare the data for `Flag_*()` by performing the specified test on the metric provided.
```

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:402]: It looks like these sections from here to the end are incomplete.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:483]: See previous discussion of `inst`.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:493]: Se previous discussion of `inst`

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:51]:
```suggestion
4.  `dfFlagged`: Flagged data; Site-level analysis results with flags added to indicate potential statistical outliers. Created by passing numeric thresholds to a `Flag_*()` function.
```

Standardizing references to function families.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:49]:
```suggestion
2.  `dfTransformed`: Transformed data; Site-level transformed data including KRI calculation. Created by `Transform_*()` functions.
```

Standardizing references to function families.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:50]:
```suggestion
3.  `dfAnalyzed`: Analyzed data; Site-level analysis results data. Created by `Analyze_*()` functions.
```

Standardizing references to function families.

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:65]:
```suggestion
4. `dfFlagged`: Flagged data; Site-level analysis results with flags added to indicate potential statistical outliers. Created by passing numeric thresholds to a `Flag_*()` function.
```

Standardizing references to function families.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Lots of small tweaks, and a couple things to discuss. Great update overall!

File: vignettes/DataPipeline.Rmd
Comment (lauramaxwell) [reply vignettes/DataPipeline.Rmd:40]: for me, this just goes to the Cookbook-Vignette page on the gsm wiki, which will need to be updated with the new cookbook (and all other vignettes) at some point, so thanks for catching that.

Comment (jonthegeek) [reply vignettes/DataPipeline.Rmd:40]: `devtools::install(build_vignettes = TRUE)` results in the proper link to a local copy of the vignette, and that blows my mind 🙃

Comment (jonthegeek) [reply vignettes/DataPipeline.Rmd:40]: I *think* this should still be updated to point to Cookbook.html directly.

```suggestion
This `data_mapping.yaml` file is provided for convenience, but may not work for all clinical studies. When no relevant data.frame is produced with the `data_mapping.yaml`, the user can use the `RunQuery()` function to write a SQL query that produces the properly mapped and filtered dataframe. Details on this can be found in the [Cookbook Vignette](Cookbook.html).
```

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:54]:
```suggestion
Each KRI has a `.yaml` file that sequentially executes all of the supporting functions and returns a list of datasets and visualizations per the specifications of the .yaml file. In the default `yaml` files included in the `system.file("workflow/metrics", package = "gsm")` directory of this package, the 5 standard datasets that are always returned are `dfTransformed`, `dfAnalyzed`, `dfFlagged`, `dfBounds`, and `dfSummary`. A step-by-step example of the data assessment workflow can be found in the [Data Analysis Vignette](DataAnalysis.html).
```

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:491]:
```suggestion
For example, the default workflow for the AE assessment (`system.file("workflow/metrics/kri0001.yaml", package = "gsm")` is shown below:
```

**Review by jonthegeek — CHANGES_REQUESTED** [review]
A few remaining tweaks for inst loading & vignette crosslinks, and this should be done!

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:163]: Should we mention an issue related to this? I want to make sure this isn't being left in accidentally (vs intentionally).

File: vignettes/DataPipeline.Rmd
Comment (jonthegeek) [inline vignettes/DataPipeline.Rmd:386]: Is this intentionally blank?

**Review by jonthegeek — APPROVED** [review]
Just a couple comments related to in-progress things. I'm approving since I'm not sure if those are intentional gaps for now, but I want to make sure they either get fixed or reported to be fixed in the future!

**Review by jonthegeek — APPROVED** [review]
Excellent! Maybe add something about the fix-reports requirement in that other PR, so this one can be closed? Or maybe a separate issue just to make sure it doesn't get lost? Or, of course, just wait until that's done to merge this one!

**Comment by jwildfire** [comment]
@jonthegeek As discussed, let's go ahead and file issues for any loose ends and get this merged today or tomorrow.

**Pull request merged**
